src/App/ExpressionParser.y

```diff
@@ -130,7 +134,11 @@ cond: exp EQ exp                                { $$ = new OperatorExpression(Do
     | '(' cond ')'                              { $$ = $2; }
     ;
 
-unit_exp: UNIT                                  { $$ = new UnitExpression(DocumentObject, $1.scaler, $1.unitStr );                }
+us_building_unit: USUNIT                        { $$ = new UnitExpression(DocumentObject, $1.scaler, $1.unitStr );                }
+other_unit: UNIT                                { $$ = new UnitExpression(DocumentObject, $1.scaler, $1.unitStr );                }
+
+unit_exp: other_unit                            { $$ = $1; }
+        | us_building_unit                      { $$ = $1; }
         | unit_exp '/' unit_exp                 { $$ = new OperatorExpression(DocumentObject, $1, OperatorExpression::DIV, $3);   }
         | unit_exp '*' unit_exp                 { $$ = new OperatorExpression(DocumentObject, $1, OperatorExpression::MUL, $3);   }

```

Comment:
On a side note, does line 143 work for anyone? I was surprised to see this in here, because I specifically remember this input not working.

I would have expected 1mm*mm to eval to 1mm^2. But it fails.

Looks like it groups as (num * unit_expr)( unit_expr) because of incorrect precedence.
NUM_AND_UNIT has higher precedence than * / and %.

Did this ever work?

Reply:
I'll have to check if this is a regression, but I would expected 1kg*m and similar to work, but currently you need to input 1kg*1m

Reply:
I just found out how to use this on master, `1 (kg*m)` works